src/ir/selector_access_path_test.cc

```diff
@@ -0,0 +1,140 @@
+#include "selector_access_path.h"
+#include "field_selector.h"
+#include "selector.h"
+#include "src/common/testing/gtest.h"
+
+#include "absl/hash/hash_testing.h"
+#include "absl/status/statusor.h"
+#include "absl/strings/str_split.h"
+
+#include <string>
+#include <vector>
+
+namespace raksha::ir {
+
+// Selector Access paths are easier to read and write in their string form, so
+// this is very useful for creating readable test inputs. This could possibly
+// be useful as a method in SelectorAccessPath itself, but, it's a bit
+// simplified right now, as it only splits on '.', the field separator
+// character. Also, I haven't found a use for it in non-test code yet.
```

Comment:
This does seem useful. Maybe by the time we have the other kinds of fields it would make more sense to use a parser-generator (like yacc, antlr, etc.) to do this easily.

Reply:
Indeed. Once you start thinking about "how would you parse an array index or a tuple access or..." I become less confident that this current implementation would scale. As such, I'm leaving it as a toy implementation in a test for now and leaving building a real implementation for later.